### DISCUSSION TITLE
Imagick output

### DISCUSSION THREAD

**Discussion by simplerethink** [lumeland/lume#356] [Q&A]
Following [the docs here](https://lume.land/plugins/imagick/), I have Imagick creating several sizes and formats. However, the output doesn't seem to generate a `picture` element to actually use the optimized sizes.

Does the plugin not generate a default `picture` like the [11ty Image plugin](https://www.11ty.dev/docs/plugins/image/#asynchronous-shortcode)? 

Could you give an example of how to modify the html output of optimized images?

**Comment by oscarotero** [comment]
Nope, Imagick plugin doesn't modify the HTML, only process the image files and output different versions (that you can use in HTML, CSS files or just as file to download).

Maybe this is an opportunity to create a `picture` plugin 🤔

**Reply by simplerethink** [reply]
Hmm, yeah, I was hoping for something like the 11ty Image plugin. I'd like to create optimized images and then have the markup rendered by the plugin.

**Reply by simplerethink** [reply]
I would think that one of the primary reasons someone would want to create multiple image sizes and formats would be to optimize images for performance and SEO, so it seems like a `picture` plugin would be a great addition.

**Reply by oscarotero** [reply]
Besides the picture element, here are many other use cases different you may want to use the Imagick plugin. For example to generate image thumbnails to use in different pages.
But yeah, a `picture` plugin to fix this concrete use case would be nice.

Maybe it could be similar to `inline` plugin, so you can convert any `<img src="" picture>` element to a `<picture>...</picture>`.

**Reply by simplerethink** [reply]
Yes, very true. 

Would the `inline` idea support other formats like `md` image format?

**Reply by oscarotero** [reply]
Yes, the [`inline`](https://lume.land/plugins/inline/) plugin is a processor, it works with the HTML generated by the templates engines (md, njk, jsx, etc). So it doesn't depend of which template engine you want to use.

The `picture` element should work in the same way.

**Reply by oscarotero** [reply]
@simplerethink This is an idea for the `picture` plugin: https://github.com/lumeland/lume/issues/384

**Reply by simplerethink** [reply]
Looks like a solid idea to me!
